### DISCUSSION TITLE
How to Set TM Parameter Timestamp Different from Packet Reception Time in YAMCS

### DISCUSSION THREAD

**Discussion by giacomogroppi** [yamcs/yamcs#950] [Q&A]
Hi everyone!
I am new to YAMCS and I am trying to configure it so that a telemetry parameter can have a timestamp that reflects the moment it was created, rather than the time when the packet containing the parameter was received.
Is there a specific way to define the timestamp for each parameter, or any method that would allow me to manage this within YAMCS?

Thank you for your help!

**Comment by xpromache** [comment]
The packet timestamp is supposed to be somehow the "creation time" of the packet - it is the job of the pre-processor to ensure that.  There is also a reception time that should be the time when the packet was received.

If you still want parameters extracted from packets with timestamp different that packet timestamps - I'm afraid it isn't possible at the moment. 

How different would be the two timestamps apart?
We may run into some issues if we were to implement such mechanism: for example when doing a replay for a time interval (including when building the Parameter Archive), Yamcs expects things to come in order and more importantly the timestamps of the parameters to be part of that time interval. Some small discrepancies may be tolerated though by using some margins.
